### DISCUSSION TITLE
Star Control I and II Collection - INSTALL.BAT crashes DOSbox

### DISCUSSION THREAD

**Discussion by Torinde** [dosbox-staging/dosbox-staging#3787] [General]
Using 0.82-bb7eb and mounting [this CD ISO](https://archive.org/details/starcontroliandiicollectiondos1995) on D:, mounting HDD image on C:

Run INSTALL.BAT - Staging closes without saying anything (I had console open and some colored line flashed briefly - would be nice to have #2183)
When doing the same in X its main window closes like Staging, but its console window remains open with:
![image](https://github.com/dosbox-staging/dosbox-staging/assets/97228894/d8587aef-2b30-4a1f-8d8e-5d18f4ce5514)

Now, even if `SET /P` came after the DOS versions in scope for Staging, it would be better if the window(s) don't just disappear, but at least stay for the user to see (e.g. show an error message box). Also, I'm not sure if real DOS on real PC crashes/reboots when encountering an invalid command in a batch file...

Not filing a bug Issue, because maybe the problem is due to some modern commands being utilized (the batch files are written by whoever packaged the ISO in 2015 or so). At the end I used _another ISO_ inside the bigger one, which was from 1995 and worked OK.

**Comment by johnnovak** [comment]
Thanks @Torinde. Yeah, that exit on a wrong command line param is total bullshit. We're slowly eradicating the way too liberal hard exits from the codebase, but we still have 201 🥶 

<img width="255" alt="image" src="https://github.com/dosbox-staging/dosbox-staging/assets/698770/32ef728e-7ebf-4b65-9422-0733f3fe1c6b">

Fix incoming soon.

**Reply by johnnovak** [reply]
Fixed here https://github.com/dosbox-staging/dosbox-staging/pull/3789

**Reply by johnnovak** [reply]
> Also, I'm not sure if real DOS on real PC crashes/reboots when encountering an invalid command in a batch file...

It doesn't 😆  That would be a crazy thing if DOS did that, man, rebooting or crashing when encountering invalid built-in command invocations 😆 DOS sucks, but it does not suck _that_ hard 😆
